refactor: re-write clipboard module test to kotlin

hurali97 opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. The three fields below are mandatory. -->

## Summary:

Migrates the **ClipboardModuleTest** Java class to Kotlin. It also changes the import for **ClipboardManager** from `android.text.ClipboardManager` to `android.content.ClipboardManager`. Since the import from `android.text` is deprecated and in the [ClipboardManager.java](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/clipboard/ClipboardModule.java) the import refers to `android.content`, so it's safe to update the import in test file as well.

It's a sub-task of the umbrella issue [here](https://github.com/facebook/react-native/issues/37708)

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->

## Changelog:

[INTERNAL] [CHANGED] ClipboardModuleTest class migrated from Java to Kotlin

<!-- Help reviewers and the release process by writing your own changelog entry.

Pick one each for the category and type tags:

[ANDROID|GENERAL|IOS|INTERNAL] [BREAKING|ADDED|CHANGED|DEPRECATED|REMOVED|FIXED|SECURITY] - Message

For more details, see:
https://reactnative.dev/contributing/changelogs-in-pull-requests
-->

## Test Plan:

🟢 Test passes when running the following:

```sh
./gradlew :packages:react-native:ReactAndroid:test
```

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes the user interface. -->

cortinico commented on packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/clipboard/ClipboardModuleTest.kt:
```diff
@@ -0,0 +1,52 @@
+@SuppressLint("ClipboardManager", "DeprecatedClass")
+@RunWith(RobolectricTestRunner::class)
+class ClipboardModuleTest {
+  private lateinit var mClipboardModule: ClipboardModule
+  private lateinit var mClipboardManager: ClipboardManager
```
```suggestion
  private lateinit var clipboardModule: ClipboardModule
  private lateinit var clipboardManager: ClipboardManager
```

hurali97 commented:
> Is missing in this PR? Did you decided to don't do it? May I ask why?

Hey I think there is some misunderstanding. What I meant was, I updated the import to reference `android.content.ClipboardManager`, so now we don't have the deprecated `import android.text.ClipboardManager` anymore. The PR was updated already !!


Also, addressed the changes for naming convention. Ready for re-review 👍

cortinico commented:
> > Is missing in this PR? Did you decided to don't do it? May I ask why?
> 
> Hey I think there is some misunderstanding. What I meant was, I updated the import to reference `android.content.ClipboardManager`, so now we don't have the deprecated `import android.text.ClipboardManager` anymore. The PR was updated already !!
> 
> Also, addressed the changes for naming convention. Ready for re-review 👍

Ah ok I understand. Sorry I misread your initial comment

cortinico reviewed (approved):
LGTM 👍
